Kudu.Services/ServiceHookHandlers/FetchHandler.cs

```diff
@@ -476,6 +498,13 @@ public static void PerformBackgroundDeployment(DeploymentInfo deployInfo, IEnvir
                     backgroundTrace.Dispose();
                 }
             });
+
+            if (waitForTempDeploymentCreation)
+            {
+                // Wait until the temp deployment is created before returning
+                var timeout = TimeSpan.FromSeconds(5);
+                await Task.WhenAny(tcs.Task, Task.Delay(timeout));
```

Comment:
After reviewing a few things, I am getting rid of the timeout. It was intended to be a safeguard against the response never being returned if something went really wrong, but now I see that the only caller that will trigger this codepath is the frontend/ARM, and its behavior heavily depends on the temp deployment being created before the response is returned.

Instead of falling back on a timeout, it will fall back on the completion of the deployment task itself, in case it blew up early or failed to run for some odd reason.